### ISSUE TITLE
Enqueue synchronization requests

### ISSUE THREAD

**Issue by stefan-niedermann** [stefan-niedermann/nextcloud-deck#554]
While using the application, there are often synchronizations performed.

- One at startup
- One after pulling down 2 refresh (reflex)
- One after creating a new card
- Background synchronization starts
- Leaving the app, go to gallery and share a file to the Deck app will again synchronize everything

This has several drawbacks, like bad device performance (it's very much activity), huge network traffic and some other impacts. While the first two will hopefully be solved with #435 i sometimes also notice SQL constraint violations when messing around with multiple parallel synchronizations as well as multiple created cards with the same title.

Those are symptoms with one cause (i think): We need to enqueue the synchronizations.

In the Notes app we are checking

1. whether or not there is already an ongoing synchronization process before starting a new one
2. whether or not there is already planned a follow-up-synchronization, so we don't run like 7 syncs sequentially

(https://github.com/stefan-niedermann/nextcloud-notes/blob/ddfd6862ad5bbd0446ec91878d06a43a28613390/app/src/main/java/it/niedermann/owncloud/notes/persistence/NoteServerSyncHelper.java#L201-L207) and while the scenario here is of course more complex, i think we should find a solution to avoid multiple parallel synchronizations.

What do you thinkg @desperateCoder ?

**Comment by desperateCoder** [comment]
I don't think we need a queue here, i guess it's fine to just prevent multiple syncs at the same time. if a sync is ongoing, what should the enqueued one sync? The things changed in the last 20 seconds? I don't think that would give us much of a benefit.

Annotation: creating a new card won't trigger a sync by its meaning, it syncs only the created card and its decendents, if given. Sure, this could have some side effects for the sync, but i think this is an absolutely edge case. I'd prefer not to prevent the user doing what he needs right now, just because a background sync is running or stuff. The side effects aren't that big as they are in two parallel whole-syncs.

Generally a pretty good idea tho, I just think the queue-part is kind of an overkill.

**Comment by stefan-niedermann** [comment]
> Annotation: creating a new card won't trigger a sync by its meaning

I know, just wanted to create a dramatic list with many many entries here :wink: 

A "queue" is too much. It's more like a "queue with maximum 2 items" - the current sync, and the next sync if items have changes while the current sync is still doing it's job.

Like i said: At least have a rough look at the link Notes application, it isn't a real "queue" there neither, it's like i described: Current + a new "request for a sync" while the current is still doing it's job.
